Makefile

```diff
@@ -2,210 +2,220 @@ ifneq (,)
-# File-lint configuration
-# -------------------------------------------------------------------------------------------------
-FL_VERSION = 0.4
-FL_IGNORES = .git/,.github/,tests/
 
 # -------------------------------------------------------------------------------------------------
-# Docker configuration
+# Default configuration
 # -------------------------------------------------------------------------------------------------
-DIR = Dockerfiles/
-FILE = Dockerfile
-IMAGE = cytopia/ansible
-TAG = latest
-NO_CACHE =
+# Own vars
+TAG        = latest
+
+# Makefile.docker overwrites
+NAME       = ansible
+VERSION    = latest
+IMAGE      = cytopia/ansible
+FLAVOUR    = default
+STAGE      = base
+DIR        = Dockerfiles
+FILE       = Dockerfile-$(STAGE)
+DOCKER_TAG = $(VERSION)
+
+ifeq ($(strip $(STAGE)),base)
+	DOCKER_TAG = $(VERSION)
+else
+	DOCKER_TAG = $(VERSION)-$(STAGE)$(KOPS)$(HELM)
+endif
 
-# Version & Flavour
-ANSIBLE = latest
-FLAVOUR = base
-KOPS =
-HELM =
+# Makefile.lint overwrites
+FL_IGNORES  = .git/,.github/,tests/
+SC_IGNORES  = .git/,.github/,tests/
+JL_IGNORES  = .git/,.github/,./tests/
 
 
 # -------------------------------------------------------------------------------------------------
-# Default Target
+#  Default Target
 # -------------------------------------------------------------------------------------------------
+.PHONY: help
 help:
 	@echo "--------------------------------------------------------------------------------"
 	@echo " Build Targets"
 	@echo "--------------------------------------------------------------------------------"
 	@echo
-	@echo "All Docker images are build as follows: $(IMAGE):\$$ANSIBLE[-\$$FLAVOUR[\$$KOPS|\$$HELM]]"
+	@echo "All Docker images are build as follows: $(IMAGE):\$$VERSION[-\$$STAGEg[\$$KOPS|\$$HELM]]"
 	@echo
-	@echo "build   [ANSIBLE=] [KOPS=] [HELM=]        Build Docker image"
-	@echo "rebuild [ANSIBLE=] [KOPS=] [HELM=]        Build Docker image without cache"
+	@echo "build   [VERSION=] [ARCH=] [KOPS=] [HELM=]        Build Docker image"
+	@echo "rebuild [VERSION=] [ARCH=] [KOPS=] [HELM=]        Build Docker image without cache"
 	@echo
-	@echo "    make build ANSIBLE=2.3"
-	@echo "    make build ANSIBLE=2.3 FLAVOUR=tools"
-	@echo "    make build ANSIBLE=2.3 FLAVOUR=infra"
-	@echo "    make build ANSIBLE=2.3 FLAVOUR=azure"
-	@echo "    make build ANSIBLE=2.3 FLAVOUR=aws"
-	@echo "    make build ANSIBLE=2.3 FLAVOUR=awsk8s"
-	@echo "    make build ANSIBLE=2.3 FLAVOUR=awshelm HELM=2.11"
-	@echo "    make build ANSIBLE=2.3 FLAVOUR=awskops KOPS=1.15"
+	@echo "    make build VERSION=2.3"
+	@echo "    make build VERSION=2.3 STAGEg=tools"
+	@echo "    make build VERSION=2.3 STAGEg=infra"
+	@echo "    make build VERSION=2.3 STAGEg=azure"
+	@echo "    make build VERSION=2.3 STAGEg=aws"
+	@echo "    make build VERSION=2.3 STAGEg=awsk8s"
+	@echo "    make build VERSION=2.3 STAGEg=awshelm HELM=2.11"
+	@echo "    make build VERSION=2.3 STAGEg=awskops KOPS=1.15"
```

Comment:
STAGEg is typo?

Reply:
No, that's the new standard from my reusable workflow. This is a "staged" build (although they are still different flavours) - meaning, they build on top of each other. I had to adjust it here or in all of my other projects.

Reply:
:facepalm: Didn't even see the `g` in your comment. Got it.